### PR TITLE
DOC: Fix warnings in doc build

### PR DESCRIPTION
**Pull request by TomAugspurger**
Split from https://github.com/pandas-dev/pandas/pull/22743

This has just the docs changes, not the doc build / CI changes, so I don't have to keep fixing merge conflicts.

### CODE REVIEW

**Comment by TomAugspurger** [comment]
Latest commit fixes a failing doctest on master

https://travis-ci.org/pandas-dev/pandas/jobs/433544423#L2688

```
2059             columns of two DataFrame objects.
2060 
2061         Notes
2062         -----
2063         If the Pearson correlation is not well defined return 'NaN'.
2064 
2065         Examples
2066         --------
2067         >>> s = pd.Series([0.25, 0.5, 0.2, -0.05])
2068         >>> s.autocorr()
Expected:
    0.1035526330902407
Got:
    0.10355263309024071
```

Assuming it's a floating point precision issue. @jorisvandenbossche @datapythonista are you OK with using `doctest: +ELLIPSIS` for this?

File: doc/source/whatsnew/v0.20.0.txt [hunk 1]
```diff
@@ -186,7 +186,7 @@ Previously, only ``gzip`` compression was supported. By default, compression of
 URLs and paths are now inferred using their file extensions. Additionally,
 support for bz2 compression in the python 2 C-engine improved (:issue:`14874`).
 
-.. ipython:: python
+.. code-block:::: python
```
Comment (datapythonista) [inline doc/source/whatsnew/v0.20.0.txt]: do we have an extra pair of colons here?

File: pandas/core/generic.py [hunk 2]
```diff
@@ -2061,9 +2061,10 @@ def to_json(self, path_or_buf=None, orient=None, date_format=None,
 
             .. versionadded:: 0.19.0
         compression : {'infer', 'gzip', 'bz2', 'zip', 'xz', None},
```
Comment (datapythonista) [inline pandas/core/generic.py]: if we remove the default, the tailing comma is not needed

File: pandas/core/series.py [hunk 3]
```diff
@@ -2065,10 +2065,10 @@ def autocorr(self, lag=1):
...
         --------
         >>> s = pd.Series([0.25, 0.5, 0.2, -0.05])
-        >>> s.autocorr()
-        0.1035526330902407
-        >>> s.autocorr(lag=2)
-        -0.9999999999999999
+        >>> s.autocorr()  # doctest: +ELLIPSIS
+        0.10355...
+        >>> s.autocorr(lag=2)  # doctest: +ELLIPSIS
+        -0.99999...
```
Comment (datapythonista) [inline pandas/core/series.py]: +1 on `doctest: +ELLIPSES`

**Review by datapythonista — COMMENTED** [review]
Great fixes, there are a couple of typos, but looks good.

**Comment by TomAugspurger** [comment]
Merging in a bit, since master is failing.

**Comment by TomAugspurger** [comment]
HTTP error downloading conda on that circle-ci one. The others passed, so merging.

**Comment by jorisvandenbossche** [comment]
Thanks a lot for this!

**Pull request merged**